File: lib/features/product/domain/entities/product.dart
import 'package:nhom2_thecoffeehouse/core/constants/enum.dart';

class Product {
  final int id;
  final String name;
  final double? price;
  final double? priceSmall;
  final double? priceMedium;
  final double? priceLarge;
  final int? categoryId;
  final String? imageUrl;
  final String? description;
  final bool isBestSeller;

  const Product({
    required this.id,
    required this.name,
    this.price,
    this.priceSmall,
    this.priceMedium,
    this.priceLarge,
    this.categoryId,
    this.imageUrl,
    this.description,
    this.isBestSeller = false,
  });

  bool hasSizeOptions() =>
      priceSmall != null || priceMedium != null || priceLarge != null;

  double? priceBySize(SizeOption? size) {
    if (size == null) return price;

    switch (size) {
      case SizeOption.small:
        return priceSmall ?? price;
      case SizeOption.medium:
        return priceMedium ?? price;
      case SizeOption.large:
        return priceLarge ?? price;
    }
  }
}

File: backend_cf/models/product.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from sqlalchemy.orm import relationship
from database import Base

class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), index=True)
    description = Column(String(255))

    price = Column(Float, nullable=True)
    price_small = Column(Float, nullable=True)
    price_medium = Column(Float, nullable=True)
    price_large = Column(Float, nullable=True)

    image_url = Column(String(255))
    category_id = Column(Integer, ForeignKey("categories.id"))

    category = relationship("Category", back_populates="products")
    favorites = relationship("Favorite", back_populates="product")

    # ✅ THÊM DÒNG NÀY
    promotion_products = relationship(
        "PromotionProduct",
        back_populates="product"
    )
